lib/main.js

```diff
@@ -47,6 +47,30 @@ const Hydrogen = {
 
     this.markerBubbleMap = {};
 
+    let skipLanguageMappingsChange = false;
+    store.subscriptions.add(
+      atom.config.onDidChange(
+        "Hydrogen.languageMappings",
+        ({ newValue, oldValue }) => {
+          if (skipLanguageMappingsChange) {
+            skipLanguageMappingsChange = false;
+            return;
+          }
+
+          if (store.runningKernels.size != 0) {
+            skipLanguageMappingsChange = true;
+
+            atom.config.set("Hydrogen.languageMappings", oldValue);
```

Comment:
Thinking about it again, this might not be so easy since we call it inside `onDidChange` and not inside `onWillChange`.